Add support for new tenant variables endpoints with environment scoping

bec-callow-oct opened the pull request:
[SC-93476] [SC-101710]

New tenant variables endpoints have been added to Octopus Server to allow multi-environment scoping for Common and Project Tenant Variables. This PR adds support for the new endpoints, and for the feature toggles endpoint to allow verification of whether the endpoints are available:

- `/api/{SpaceId}/tenants/{TenantId}/commonvariables` (GET/PUT/POST)
- `/api/{SpaceId}/tenants/{TenantId}/projectvariables` (GET/PUT/POST)

Feature toggles can be queried by name - both of the above endpoints are under `CommonVariableScopingFeatureToggle`.

tothegills commented on pkg/tenants/tenant_service.go:
```diff
@@ -314,3 +314,36 @@ func DeleteByID(client newclient.Client, spaceID string, ID string) error {
+}
+
+// UpdateCommonVariables modifies tenant common variables based on the ones provided as input.
+func UpdateCommonVariables(client newclient.Client, spaceID string, ID string, commonVariables *variables.ModifyTenantCommonVariablesCommand) (*variables.TenantCommonVariablesResource, error) {
```
What is `ID` in this context? Is it the tenant id? There is also a tenant id on `ModifyTenantCommonVariablesCommand`.

bec-callow-oct replied:
Renamed to TenantId and removed that field from the command

tothegills reviewed (commented):
Confirmed that getting feature toggles, tenant project variables and tenant common variables are all working as expected.

I'm struggling to use the update commands, have an example?
```
	var vars []variables.TenantCommonVariableCommand
	modify := variables.ModifyTenantCommonVariablesCommand{TenantID: tenantID, CommonVariables: vars}
	modified, err := tenants.UpdateCommonVariables(client, "Spaces-1", tenantID, &modify)
	if err != nil {
		_ = fmt.Errorf("error creating API client: %v", err)
		return
	}
	fmt.Println(modified)
```

bec-callow-oct commented:
> Confirmed that getting feature toggles, tenant project variables and tenant common variables are all working as expected.
> 
> I'm struggling to use the update commands, have an example?
> 
> ```
> 	var vars []variables.TenantCommonVariableCommand
> 	modify := variables.ModifyTenantCommonVariablesCommand{TenantID: tenantID, CommonVariables: vars}
> 	modified, err := tenants.UpdateCommonVariables(client, "Spaces-1", tenantID, &modify)
> 	if err != nil {
> 		_ = fmt.Errorf("error creating API client: %v", err)
> 		return
> 	}
> 	fmt.Println(modified)
> ```

I've made a few changes and verified that update is partially working. For now, it can be tested as follows but an empty list of variables is causing a server error:

```
vars := []variables.TenantCommonVariableCommand{
		{
		        Id: "<Variable ID (optional)>"
			LibraryVariableSetId: "<LibraryVariableSet ID>",
			TemplateID:           "<Template ID>",
			Value: core.PropertyValue{
				IsSensitive: false,
				Value:       "New value from Go client",
			},
			Scope: variables.TenantVariableScope{
				EnvironmentIds: []string{
					"<Environment ID>",
				},
			},
		},
		}

		modify := variables.ModifyTenantCommonVariablesCommand{Variables: vars}
		modified, err := tenants.UpdateCommonVariables(client, "Spaces-1", tenant.ID, &modify)

		fmt.Println(modified)
		fmt.Println(err)
```

I'm currently investigating why I'm getting `Octopus API error: There was a problem with your request. [The Variables field is required.]` when variables is empty as I've removed `omitempty`. Will update when I have a solution

bec-callow-oct commented on pkg/variables/tenant_project_variable.go:
```diff
@@ -0,0 +1,47 @@
+	SpaceID  string `json:"SpaceId"`
+}
+
+type TenantProjectVariablesResource struct {
```
Rather than `TenantProjectVariablesResource`, how about `TenantProjectVariableResponse`? This would help to make it a bit clearer what the object is intended to be used for. Within that, we currently have `TenantProjectVariable`, which could be renamed to something similar to `TenantProjectVariableValueResponse` or `TenantProjectVariableDetailsResponse`

bec-callow-oct replied:
Names updated based on the team discussion

tothegills commented:
Ran into an issue because I used `TenantProjectVariableCommand.ID` instead of `TenantProjectVariableCommand.Id`. Why are there both?

tothegills commented on pkg/tenants/tenant_service.go:
```diff
@@ -314,3 +314,26 @@ func DeleteByID(client newclient.Client, spaceID string, ID string) error {
+}
+
+// UpdateProjectVariables modifies tenant project variables based on the ones provided as input.
+func UpdateProjectVariables(client newclient.Client, spaceID string, tenantID string, projectVariables *variables.ModifyTenantProjectVariablesCommand) (*variables.TenantProjectVariablesResource, error) {
```
✔️

tothegills reviewed (changes requested):
Some naming to address, leaving `Resource` in the past.
The methods all work as expected.
I'm wondering about `Id` vs `ID` and the resource properties on the structs (`resources.Resource`), do we need those?

bec-callow-oct commented:
> Some naming to address, leaving `Resource` in the past. The methods all work as expected. I'm wondering about `Id` vs `ID` and the resource properties on the structs (`resources.Resource`), do we need those?

I've removed the extra IDs and updated the naming
